gpt_json/parsers.py

```diff
@@ -35,3 +37,76 @@ def find_json_response(full_response, extract_type):
         extracted_response = extracted_response.group(0)
 
     return extracted_response
+
+
+def _is_valid_json(substring: str) -> bool:
+    try:
+        json.loads(substring)
+        return True
+    except json.decoder.JSONDecodeError:
+        return False
+
+def _fix_broken_json(substring: str) -> tuple[str, str]:
```

Comment:
This logic duplicates some portion of what is implemented in `gpt_json.transformations.fix_truncated_json`. Instead of reinventing the wheel here, I'd suggest adding additional coverage and functionality to the parser within `build_stack`.

Specifically - based on a brief skim here - I think you'll want to:
- Detect if we are before an unescaped colon or after, which indicates that we are adding a key
- After a colon but before a comma or closing bracket `}` which indicates that the value is still pending. This will let you wait until full integer fields have been yielded.
- Aside from that you write a sibling function to `fix_truncated_json` that will also inject valid JSON values like `null` if no value is yet supplied

Reply:
ah didn't notice the overlap, this is sick. your `build_stack` is way more general too lol